src/keywrap/keyprovider.rs

```diff
@@ -135,6 +142,44 @@ impl KeyProviderKeyWrapProtocolOutput {
         serde_json::from_slice(&resp_bytes)
             .map_err(|_| anyhow!("keyprovider: failed to deserialize message from binary executor"))
     }
+
+    async fn from_ttrpc(input: Vec<u8>, conn: &str, operation: OpKey) -> Result<Self> {
+        let c = Client::connect(conn).unwrap();
+
+        let kc = keyprovider_ttrpc::KeyProviderServiceClient::new(c);
+        let kc1 = kc.clone();
+        let mut req = keyprovider_ttrpc_structs::keyProviderKeyWrapProtocolInput::new();
+        req.set_KeyProviderKeyWrapProtocolInput(input);
+
+        let ttrpc_output = match operation {
+            OpKey::Wrap => kc1
+                .wrap_key(context::with_timeout(20 * 1000 * 1000), &req)
```

Comment:
Could you define a global `const` for this duplicated timeout value (with a name that shows the units for this value).